src/Gui/SoDatumLabel.cpp

```diff
@@ -613,10 +688,27 @@ SbVec3f SoDatumLabel::getLabelTextCenterArcLength(const SbVec3f& ctr, const SbVe
 {
     float length = this->param1.getValue();
 
-            // Text location
+    // Angles calculations
+    SbVec3f vc1 = (p1 - ctr);
```

Comment:
Don't repeat yourself! The exact same code block is used above. Consider refactoring this code part.

Reply:
Above is on the `DatumLabelBox` class, here we are in `SoDatumLabel`.

Notice the whole file needs refactoring. Maybe out of scope of this PR even if I already start it.
We could create `InfoArcAngle`, `InfoArcLength` - and so ones - classes to give all the values required multiple times.